feature/notification/src/main/java/io/github/droidkaigi/confsched2019/notification/service/ManageTopicSubscriptionIntentService.kt

```diff
@@ -0,0 +1,163 @@
+package io.github.droidkaigi.confsched2019.notification.service
+
+import android.annotation.SuppressLint
+import android.app.AlarmManager
+import android.app.IntentService
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.AlarmManagerCompat
+import androidx.core.content.ContextCompat
+import com.google.firebase.iid.FirebaseInstanceId
+import com.google.firebase.messaging.FirebaseMessaging
+import io.github.droidkaigi.confsched2019.notification.NotificationChannelInfo
+import io.github.droidkaigi.confsched2019.notification.R
+import io.github.droidkaigi.confsched2019.notification.Topic
+import io.github.droidkaigi.confsched2019.notification.notificationBuilder
+import io.github.droidkaigi.confsched2019.timber.error
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.tasks.await
+import timber.log.Timber
+import timber.log.debug
+import java.util.concurrent.TimeUnit
+
+class ManageTopicSubscriptionIntentService : IntentService(NAME) {
```

Comment:
Probably can we create an issue that uses work manager instead of IntentService?

Reply:
WorkManager should be used if the job depends on some conditions. As for this service, that's a ... maybe *the network state* if say anything?

Reply:
I can also write conditions. I think WorkManager is more simple and that can retry with backoff. It also supports coroutines.
I think that it is convenient when there is a task you want to complete.

Reply:
But I think that there is no need to do it now.

Reply:
Indeed.

https://github.com/DroidKaigi/conference-app-2019/pull/490/files#diff-81011ed4929c2dc6f91379c063a1906eR38 can complete this task across app processes though, retry-feature looks simple than self-retrying. Anyway, we can delegate the migration to WorkManager to anyone.